djangocms_navigation/models.py

```diff
@@ -0,0 +1,45 @@
+from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.sites.models import Site
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from treebeard.mp_tree import MP_Node
+
+
+class Menu(models.Model):
+    """
+    MenuContent Grouper
+    """
+    pass
+
+
+class MenuContent(models.Model):
+    title = models.CharField(verbose_name=_('title'), max_length=100)
+    site = models.ForeignKey(
```

Comment:
I still don't know why this relation is here and not in Menu. Would it mean you can just switch an existing menu to a different site in a different version? Does it mean you can also have two published menu contents that point to a single site?

Reply:
My thinking was, a single site can have multiple menus. I raised question with Noel to confirm

Reply:
I think site and menu identifier should be in Menu then, and they should be uneditable/unique_together.

Reply:
That's assuming we shouldn't be able to change the site after adding a menu (or if we do change it then the change should not be versioned). It would make sense to me that menus shouldn't suddenly change sites, but I'm not sure what the requirement is tbh.
And my understanding is the same as Riz's - that we can have multiple menus on one site. So for example there could be a top nav menu at the top which maybe all pages would have, but then a sidebar menu which changes depending on the section of the site we're in.

Reply:
Now I am thinking it and I agree site should be in Menu however it shouldn't be unique_together to allow multiple. For example, if user created menu under one site and plugin render will use that particular site and all the node added to generate URL but if then site change then all node URL won't work as they belong to another site, right?

Reply:
That's why you get unique_together on (site, identifier). This way you can't have two menus with the same identifier on the same site.